tiatoolbox/tools/pyramid.py

```diff
@@ -0,0 +1,526 @@
+                resolution=1 / scale,
+                units="baseline",
+                pad_mode=pad_mode,
+                interpolation=interpolation,
+            )
+        return Image.fromarray(rgb)
+
+    def tile_path(self, level: int, x: int, y: int) -> Path:
+        """Generate the path for a specified tile.
+
+        Args:
+            level (int):
+                The pyramid level of the tile starting from 0
+                (the whole slide in one tile, 0-0-0).
+            x (int):
+                The tile index in the x direction.
+            y (int):
+                The tile index in the y direction.
+
+        Returns:
+            Path: A pathlib path object with two parts.
+
+        """
+        raise NotImplementedError
+
+    def dump(self, path: Union[str, Path], container=None, compression=None):
+        """Write all tiles to disk.
+
+        Arguments:
+            path (str or Path)
+            container (str): Container to use. Defaults to None which
+                saves to a directory. Possible values are "zip", "tar".
+            compression (str): Compression method. Defaults to None.
+                Possible values are None, "deflate", "gzip",
+                "bz2", "lzma". Note that tar does not support deflate
+                and zip does not support gzip.
+        """
+        path = Path(path)
+        if container == "zip":
+            compression2enum = {
+                None: zipfile.ZIP_STORED,
+                "deflate": zipfile.ZIP_DEFLATED,
+                "bz2": zipfile.ZIP_BZIP2,
+                "lzma": zipfile.ZIP_LZMA,
+            }
+            if compression not in compression2enum:
+                raise ValueError("Unsupported compression for zip")
+            archive = zipfile.ZipFile(
+                path, mode="w", compression=compression2enum[compression]
+            )
+        elif container == "tar":
+            compression2mode = {
+                None: "w",
+                "gzip": "w:gz",
+                "bz2": "w:bz2",
+                "lzma": "w:xz",
+            }
+            if compression not in compression2mode:
+                raise ValueError("Unsupported compression for tar")
+
+            archive = tarfile.TarFile.open(path, mode=compression2mode[compression])
+        elif container is None:
+            path.mkdir(parents=False)
+            if compression is not None:
+                raise ValueError("Unsupported compression for container None")
+        else:
+            raise ValueError("Unsupported container")
+        for level in range(self.level_count):
+            for x, y in np.ndindex(self.tile_grid_size(level)):
+                tile = self.get_tile(level=level, x=x, y=y)
+                tile_path = self.tile_path(level, x, y)
+                full_path = path / tile_path
+                if container is None:
+                    full_path.parent.mkdir(parents=True, exist_ok=True)
+                if container is None:
```

Comment:
Why are there two "if container is None:" statements ?

Reply:
No reason. Looks like a mistake. Good spot.